// ngx-newsticker-albe/__ivy_ngcc__/fesm2015/ngx-newsticker-albe.js
import { Component, ElementRef, Renderer2, Input, ViewChild, ViewChildren, HostListener, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

const _c0 = ["nt"];
const _c1 = ["ntCounter"];
const _c2 = ["ntTitle"];
const _c3 = ["nDart"];
const _c4 = ["ntNavi"];
function NgxNewstickerAlbeComponent_small_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "small", 13, 14);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.current);
} }
class NgxNewstickerAlbeComponent {
    /**
     * @param {?} elementRef
     * @param {?} renderer
     */
    constructor(elementRef, renderer) {
        this.elementRef = elementRef;
        this.renderer = renderer;
        // Highlighted text.
        this.title = '';
        // List of messages to be displayed.
        this.events = [];
        // Set the time interval between the text exchange.
        this.interval = 3000;
        // Sets the current count visibility.
        this.showCounter = true;
        this.ACTIONS = {
            'NEXT': 1,
            'PREV': -1
        };
        this.position = -1;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.autoNavigate();
        this.navigate('NEXT'); // initialize
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        this.defaultColor = this.defaultColor || '#1976D2';
        // this.elementRef.nativeElement.style.backgroundColor = this.backgroundColor;
        this.renderer.setStyle(this.ntRef.nativeElement, 'background-color', this.backColor || '#FFFFFF');
        this.renderer.setStyle(this.ntRef.nativeElement, 'border-color', this.defaultColor);
        if (this.ntCounterRef) {
            this.renderer.setStyle(this.ntCounterRef.nativeElement, 'background-color', this.defaultColor);
        }
        if (this.ntTitleRef) {
            this.renderer.setStyle(this.ntTitleRef.nativeElement, 'background-color', this.defaultColor);
            this.renderer.setStyle(this.nDart.nativeElement, 'border-left-color', this.defaultColor);
        }
        this.ntNaviRef.forEach((/**
         * @param {?} item
         * @return {?}
         */
        item => {
            this.renderer.setStyle(item.nativeElement, 'border-color', this.defaultColor);
        }));
    }
    /**
     * @return {?}
     */
    onMouseOver() {
        clearInterval(this.autoNext);
    }
    /**
     * @return {?}
     */
    onMouseLeave() {
        this.autoNavigate();
    }
    /**
     * @private
     * @return {?}
     */
    autoNavigate() {
        if (this.interval > 0) {
            this.autoNext = setInterval((/**
             * @return {?}
             */
            () => this.navigate('NEXT')), this.interval);
        }
        else {
            clearInterval(this.autoNext);
        }
    }
    /**
     * @param {?} action
     * @return {?}
     */
    navigate(action) {
        if (typeof this.ACTIONS[action] === 'undefined') {
            return;
        }
        /** @type {?} */
        const shift = this.ACTIONS[action];
        this.position = this.position + shift;
        if (this.events) {
            if (this.position < 0) {
                this.position = this.events.length - 1;
            }
            else if (this.position >= this.events.length) {
                this.position = 0;
            }
            this.item = this.events[this.position];
            this.current = `${(this.position + 1)}/${this.events.length}`;
        }
    }
}
NgxNewstickerAlbeComponent.ɵfac = function NgxNewstickerAlbeComponent_Factory(t) { return new (t || NgxNewstickerAlbeComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2)); };
NgxNewstickerAlbeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgxNewstickerAlbeComponent, selectors: [["ngx-newsticker"]], viewQuery: function NgxNewstickerAlbeComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true);
        ɵngcc0.ɵɵstaticViewQuery(_c1, true);
        ɵngcc0.ɵɵstaticViewQuery(_c2, true);
        ɵngcc0.ɵɵstaticViewQuery(_c3, true);
        ɵngcc0.ɵɵviewQuery(_c4, true);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.ntRef = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.ntCounterRef = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.ntTitleRef = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.nDart = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.ntNaviRef = _t);
    } }, hostBindings: function NgxNewstickerAlbeComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("mouseover", function NgxNewstickerAlbeComponent_mouseover_HostBindingHandler() { return ctx.onMouseOver(); })("mouseleave", function NgxNewstickerAlbeComponent_mouseleave_HostBindingHandler() { return ctx.onMouseLeave(); });
    } }, inputs: { title: "title", events: "events", interval: "interval", showCounter: "showCounter", defaultColor: "defaultColor", backColor: "backColor" }, decls: 18, vars: 3, consts: [[1, "newsticker"], ["nt", ""], ["class", "nt-counter", 4, "ngIf"], [1, "nt-title"], ["ntTitle", ""], ["nDart", ""], [1, "nt-content"], [3, "innerHTML"], [1, "nt-navi"], [3, "click"], [1, "arrow-left"], ["ntNavi", ""], [1, "arrow-right"], [1, "nt-counter"], ["ntCounter", ""]], template: function NgxNewstickerAlbeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0, 1);
        ɵngcc0.ɵɵtemplate(2, NgxNewstickerAlbeComponent_small_2_Template, 3, 1, "small", 2);
        ɵngcc0.ɵɵelementStart(3, "div", 3, 4);
        ɵngcc0.ɵɵelementStart(5, "h2");
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(7, "span", null, 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 6);
        ɵngcc0.ɵɵelement(10, "label", 7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 8);
        ɵngcc0.ɵɵelementStart(12, "span", 9);
        ɵngcc0.ɵɵlistener("click", function NgxNewstickerAlbeComponent_Template_span_click_12_listener() { return ctx.navigate("PREV"); });
        ɵngcc0.ɵɵelement(13, "i", 10, 11);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(15, "span", 9);
        ɵngcc0.ɵɵlistener("click", function NgxNewstickerAlbeComponent_Template_span_click_15_listener() { return ctx.navigate("NEXT"); });
        ɵngcc0.ɵɵelement(16, "i", 12, 11);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showCounter);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate(ctx.title);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("innerHTML", ctx.item, ɵngcc0.ɵɵsanitizeHtml);
    } }, directives: [ɵngcc1.NgIf], styles: [".newsticker[_ngcontent-%COMP%]{width:100%;height:40px;background-color:#fff;position:relative;border:1px solid #1976d2}.newsticker[_ngcontent-%COMP%] > .nt-counter[_ngcontent-%COMP%]{position:absolute;top:-20px;right:-1px;padding:5px;background-color:#1976d2;color:#fff}.newsticker[_ngcontent-%COMP%] > .nt-title[_ngcontent-%COMP%]{width:auto;max-width:180px;height:40px;display:inline-block;background-color:#1976d2;position:relative;text-transform:uppercase}.newsticker[_ngcontent-%COMP%] > .nt-title[_ngcontent-%COMP%] > h2[_ngcontent-%COMP%]{display:inline-block;margin:0;padding:0 15px;line-height:40px;font-size:18px;color:#fff;height:40px;box-sizing:border-box;overflow:hidden;font-weight:lighter}.newsticker[_ngcontent-%COMP%] > .nt-title[_ngcontent-%COMP%] > span[_ngcontent-%COMP%]{width:0;position:absolute;right:-10px;top:10px;height:0;border-style:solid;border-width:10px 0 10px 10px;border-color:transparent transparent transparent #1976d2}.newsticker[_ngcontent-%COMP%] > .nt-content[_ngcontent-%COMP%]{padding:0;margin:0;position:absolute;top:0;left:200px;right:40px;height:40px;font-size:16px}.newsticker[_ngcontent-%COMP%] > .nt-content[_ngcontent-%COMP%] > label[_ngcontent-%COMP%]{position:absolute;height:40px;width:100%;line-height:40px;text-decoration:none;color:#333;overflow:hidden;display:block;white-space:nowrap;text-overflow:ellipsis;font-weight:400}.newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%]{width:50px;height:40px;position:absolute;right:0;top:0;opacity:.1}.newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%] > span[_ngcontent-%COMP%]{width:25px;height:40px;position:absolute;cursor:pointer}.newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%] > span[_ngcontent-%COMP%]:first-child{left:0}.newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%] > span[_ngcontent-%COMP%]:last-child{right:0}.newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%] > span[_ngcontent-%COMP%] > i[_ngcontent-%COMP%]{border:solid #1976d2;border-width:0 2px 2px 0;display:inline-block;padding:6px;position:absolute;top:13px;opacity:.5}.newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%] > span[_ngcontent-%COMP%] > i.arrow-right[_ngcontent-%COMP%]{transform:rotate(-45deg);-webkit-transform:rotate(-45deg)}.newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%] > span[_ngcontent-%COMP%] > i.arrow-left[_ngcontent-%COMP%]{transform:rotate(135deg);-webkit-transform:rotate(135deg);left:10px}.newsticker[_ngcontent-%COMP%]:hover   .nt-navi[_ngcontent-%COMP%]{opacity:1}.newsticker[_ngcontent-%COMP%] > .nt-content[_ngcontent-%COMP%] > label[_ngcontent-%COMP%], .newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%], .newsticker[_ngcontent-%COMP%] > .nt-navi[_ngcontent-%COMP%] > span[_ngcontent-%COMP%]{transition:.25s linear;-moz-transition:.25s linear;-webkit-transition:.25s linear}"] });
/** @nocollapse */
NgxNewstickerAlbeComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NgxNewstickerAlbeComponent.propDecorators = {
    title: [{ type: Input }],
    events: [{ type: Input }],
    interval: [{ type: Input }],
    showCounter: [{ type: Input }],
    defaultColor: [{ type: Input }],
    backColor: [{ type: Input }],
    ntRef: [{ type: ViewChild, args: ['nt', { static: true },] }],
    ntCounterRef: [{ type: ViewChild, args: ['ntCounter', { static: true },] }],
    ntTitleRef: [{ type: ViewChild, args: ['ntTitle', { static: true },] }],
    nDart: [{ type: ViewChild, args: ['nDart', { static: true },] }],
    ntNaviRef: [{ type: ViewChildren, args: ['ntNavi',] }],
    onMouseOver: [{ type: HostListener, args: ['mouseover',] }],
    onMouseLeave: [{ type: HostListener, args: ['mouseleave',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxNewstickerAlbeComponent, [{
        type: Component,
        args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ngx-newsticker',
                template: "<div #nt class=\"newsticker\">\r\n  <small #ntCounter class=\"nt-counter\" *ngIf=\"showCounter\">{{current}}</small>\r\n  <div #ntTitle class=\"nt-title\">\r\n    <h2>{{title}}</h2>\r\n    <span #nDart></span>\r\n  </div>\r\n  <div class=\"nt-content\">\r\n    <label [innerHTML]=\"item\"></label>\r\n  </div>\r\n  <div class=\"nt-navi\">\r\n    <span (click)=\"navigate('PREV')\">\r\n      <i #ntNavi class=\"arrow-left\"></i>\r\n    </span>\r\n    <span (click)=\"navigate('NEXT')\">\r\n      <i #ntNavi class=\"arrow-right\"></i>\r\n    </span>\r\n  </div>\r\n</div>\r\n",
                styles: [".newsticker{width:100%;height:40px;background-color:#fff;position:relative;border:1px solid #1976d2}.newsticker>.nt-counter{position:absolute;top:-20px;right:-1px;padding:5px;background-color:#1976d2;color:#fff}.newsticker>.nt-title{width:auto;max-width:180px;height:40px;display:inline-block;background-color:#1976d2;position:relative;text-transform:uppercase}.newsticker>.nt-title>h2{display:inline-block;margin:0;padding:0 15px;line-height:40px;font-size:18px;color:#fff;height:40px;box-sizing:border-box;overflow:hidden;font-weight:lighter}.newsticker>.nt-title>span{width:0;position:absolute;right:-10px;top:10px;height:0;border-style:solid;border-width:10px 0 10px 10px;border-color:transparent transparent transparent #1976d2}.newsticker>.nt-content{padding:0;margin:0;position:absolute;top:0;left:200px;right:40px;height:40px;font-size:16px}.newsticker>.nt-content>label{position:absolute;height:40px;width:100%;line-height:40px;text-decoration:none;color:#333;overflow:hidden;display:block;white-space:nowrap;text-overflow:ellipsis;font-weight:400}.newsticker>.nt-navi{width:50px;height:40px;position:absolute;right:0;top:0;opacity:.1}.newsticker>.nt-navi>span{width:25px;height:40px;position:absolute;cursor:pointer}.newsticker>.nt-navi>span:first-child{left:0}.newsticker>.nt-navi>span:last-child{right:0}.newsticker>.nt-navi>span>i{border:solid #1976d2;border-width:0 2px 2px 0;display:inline-block;padding:6px;position:absolute;top:13px;opacity:.5}.newsticker>.nt-navi>span>i.arrow-right{transform:rotate(-45deg);-webkit-transform:rotate(-45deg)}.newsticker>.nt-navi>span>i.arrow-left{transform:rotate(135deg);-webkit-transform:rotate(135deg);left:10px}.newsticker:hover .nt-navi{opacity:1}.newsticker>.nt-content>label,.newsticker>.nt-navi,.newsticker>.nt-navi>span{transition:.25s linear;-moz-transition:.25s linear;-webkit-transition:.25s linear}"]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }]; }, { title: [{
            type: Input
        }], events: [{
            type: Input
        }], interval: [{
            type: Input
        }], showCounter: [{
            type: Input
        }], defaultColor: [{
            type: Input
        }], 
    /**
     * @return {?}
     */
    onMouseOver: [{
            type: HostListener,
            args: ['mouseover']
        }], 
    /**
     * @return {?}
     */
    onMouseLeave: [{
            type: HostListener,
            args: ['mouseleave']
        }], backColor: [{
            type: Input
        }], ntRef: [{
            type: ViewChild,
            args: ['nt', { static: true }]
        }], ntCounterRef: [{
            type: ViewChild,
            args: ['ntCounter', { static: true }]
        }], ntTitleRef: [{
            type: ViewChild,
            args: ['ntTitle', { static: true }]
        }], nDart: [{
            type: ViewChild,
            args: ['nDart', { static: true }]
        }], ntNaviRef: [{
            type: ViewChildren,
            args: ['ntNavi']
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgxNewstickerAlbeModule {
}
NgxNewstickerAlbeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxNewstickerAlbeModule });
NgxNewstickerAlbeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxNewstickerAlbeModule_Factory(t) { return new (t || NgxNewstickerAlbeModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxNewstickerAlbeModule, { declarations: function () { return [NgxNewstickerAlbeComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxNewstickerAlbeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxNewstickerAlbeModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [NgxNewstickerAlbeComponent],
                exports: [NgxNewstickerAlbeComponent]
            }]
    }], null, null); })();

export { NgxNewstickerAlbeComponent, NgxNewstickerAlbeModule };
